### PR TITLE
selectively propagate debugging arguments for payload

### LINKED ISSUE #95957: Can't open a folder when extensionHost debugging in web
Happens with both extensionHost and pwa-extensionHost.
Repro steps:
Open a folder with web (I used the pull request extension)
F5
A new tab opens with the extension host
Open a folder in the new tab

Expected: Still extensionHost debugging in the new tab
Actual: It just opens the folder in web VS Code, no extensionHost debugging.

### PR DESCRIPTION
**Pull request by weinand**
This PR fixes #95957

The fix analyses the properties in the current payload and copies only debugging related properties to a new payload which is passed to the newly opened tab.
If no debugging related properties are found no payload is passed.

This fix works for the "Open folder" and the "Open workspace" case.

### CODE REVIEW

File: src/vs/workbench/services/host/browser/browserHostService.ts [hunk 1]
```diff
@@ -117,14 +117,34 @@ export class BrowserHostService extends Disposable implements IHostService {
 			const openable = toOpen[i];
 			openable.label = openable.label || this.getRecentLabel(openable);
 
+			// selectively copy payload: for now only extension debugging properties are considered
+			const originalPayload = this.workspaceProvider.payload;
+			let newPayload: Array<any> | undefined = undefined;
```
Comment (bpasero) [inline src/vs/workbench/services/host/browser/browserHostService.ts:122]:
```suggestion
			let newPayload: Array<unknown> | undefined = undefined;
```

**Review by bpasero — CHANGES_REQUESTED** [review]
Thanks! 

I wonder though, is it really necessary to parse the `payload` from the workspace provider if the properties are already existing on the `IWorkbenchEnvironmentService`? Can we not simply take them from there?

**Comment by weinand** [comment]
@bpasero Yes, I'll investigate whether getting the properties from the `IWorkbenchEnvironmentService` is straightforward. On first glimpse it was not...

**Comment by weinand** [comment]
Since I'm running out of time, I will release the current PR and postpone @bpasero's suggestion to debt week.

**Comment by bpasero** [comment]
@weinand sounds good, can you open an issue to remember?

**Comment by weinand** [comment]
done: #96337

**Pull request merged**